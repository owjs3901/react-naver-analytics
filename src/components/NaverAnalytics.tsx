/**
 * Naver Analytics Script with initial configuration
 * @constructor
 */
export function NaverAnalytics({ accountId }: { accountId: string }) {
  return (
    <>
      <script src="//wcs.naver.net/wcslog.js" type="text/javascript"></script>
      <script
        dangerouslySetInnerHTML={{
          __html: `if(!wcs_add)var wcs_add={};wcs_add["wa"]="${accountId}";if(window.wcs)wcs_do();`,
        }}
        type="text/javascript"
      />
    </>
  );
}
