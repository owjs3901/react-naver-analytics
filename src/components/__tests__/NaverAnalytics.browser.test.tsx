import { render } from "@testing-library/react";
import { NaverAnalytics } from "../NaverAnalytics";

describe("NaverAnalytics", () => {
  it("render script", () => {
    const { container } = render(<NaverAnalytics accountId="mock" />);
    expect(container).toMatchSnapshot();
  });
});
