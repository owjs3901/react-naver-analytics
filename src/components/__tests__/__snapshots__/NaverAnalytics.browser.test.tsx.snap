// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NaverAnalytics > render script 1`] = `
<div>
  <script
    src="//wcs.naver.net/wcslog.js"
    type="text/javascript"
  >
    
  </script>
  <script
    type="text/javascript"
  >
    if(!wcs_add)var wcs_add={};wcs_add["wa"]="mock";if(window.wcs)wcs_do();
  </script>
</div>
`;
